From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 4 Feb 2014 16:55:06 +0100
Subject: [PATCH] opkg-build: add control.tar.gz before data.tar.gz

This is not strictly necessary for opkg. However, dpkg now requieres this
and it can be convenient to use it (e.g. dpkg --contents <something>.ipk).

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 opkg-build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/opkg-build b/opkg-build
index 646787b..68ca5e2 100755
--- a/opkg-build
+++ b/opkg-build
@@ -250,9 +250,9 @@ if [ "$outer" = "ar" ] ; then
   # chown the content to "0:0".  files. This is needed as high UID values (>
   # 99999) cause problems when parsing ar file headers
   ( cd $tmp_dir && chown 0:0 ./debian-binary ./data.tar.gz ./control.tar.gz &&
-	${CROSS_COMPILE}ar -crf $pkg_file ./debian-binary ./data.tar.gz ./control.tar.gz )
+	${CROSS_COMPILE}ar -crf $pkg_file ./debian-binary ./control.tar.gz ./data.tar.gz )
 else
-  ( cd $tmp_dir && tar -cz --format=gnu -f $pkg_file ./debian-binary ./data.tar.gz ./control.tar.gz )
+  ( cd $tmp_dir && tar -cz --format=gnu -f $pkg_file ./debian-binary ./control.tar.gz ./data.tar.gz )
 fi
 
 rm $tmp_dir/debian-binary $tmp_dir/data.tar.gz $tmp_dir/control.tar.gz
